### PR TITLE
TerrainPhysicsCollider docs.

### PR DESCRIPTION
**Pull request by sphrose**

### CODE REVIEW

**Comment by willihay** [comment]
Docs for the "Terrain Physics Collider component" seem like they should go in the Component Reference section of the User Guide, not the Gem Reference.

**Comment by sphrose** [comment]
> 
> 
> Docs for the "Terrain Physics Collider component" seem like they should go in the Component Reference section of the User Guide, not the Gem Reference.

I'm happy to move it, but the component is part of the terrain gem, so won't be accessible without it.

**Comment by willihay** [comment]
> > Docs for the "Terrain Physics Collider component" seem like they should go in the Component Reference section of the User Guide, not the Gem Reference.
> 
> I'm happy to move it, but the component is part of the terrain gem, so won't be accessible without it.

Yup, that's just how we have the User Guide organized. Similar situation with PhysX Gems and components. We have a Gem reference section and a component reference section. Component reference docs should go under /user-guide/components/reference. You can create a terrain subdirectory there. If there's only 1 terrain component at the moment, then you can name the file _index.md.

**Comment by micronAMZN** [comment]
> 
> 
> > > Docs for the "Terrain Physics Collider component" seem like they should go in the Component Reference section of the User Guide, not the Gem Reference.
> > 
> > 
> > I'm happy to move it, but the component is part of the terrain gem, so won't be accessible without it.
> 
> Yup, that's just how we have the User Guide organized. Similar situation with PhysX Gems and components. We have a Gem reference section and a component reference section. Component reference docs should go under /user-guide/components/reference. You can create a terrain subdirectory there. If there's only 1 terrain component at the moment, then you can name the file _index.md.

Related to  #860

The gems/reference/terrain/_index.md should be:
content/docs/user-guide/gems/reference/environments/terrain.md

The component should be:
content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md

content/docs/user-guide/gems/reference/environments/terrain.md link to component should be updated.
content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md link to gem should be updated.

File: content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+title: Terrain Physics Collider component
+linktitle: TerrainPhysicsCollider
+description: ' Open 3D Engine (O3DE) Terrain Physics Collider reference. '
+weight: 100
+---
+
+The **Terrain Physics Collider** component provides terrain data to the physics system in the form of a heightfield.  
+The dimensions of the collider can be configured by modifying the [Axis Aligned Box Shape Component](/docs/user-guide/components/reference/shape/axis-aligned-box-shape) on the same entity.
```
Comment (JwGedit) [inline content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md:9]:
```suggestion
You can configure the dimensions of the collider by modifying the [Axis Aligned Box Shape Component](/docs/user-guide/components/reference/shape/axis-aligned-box-shape) on the same entity.
```

**Review by JwGedit — CHANGES_REQUESTED** [review]

File: content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md (hunk 1 above)
Comment (sphrose) [reply content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md:9]: Changed.

File: content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+---
+title: Terrain Physics Collider component
+linktitle: TerrainPhysicsCollider
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md:3]:
```suggestion
linkTitle: Terrain Physics Collider
```

File: content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+---
+title: Terrain Physics Collider component
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md:2]:
```suggestion
title: Terrain Physics Collider Component
```

File: content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+linktitle: TerrainPhysicsCollider
+description: ' Open 3D Engine (O3DE) Terrain Physics Collider reference. '
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md:4]:
```suggestion
description: Open 3D Engine (O3DE) Terrain Physics Collider component reference.
```

File: content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+weight: 100
+---
+
+The **Terrain Physics Collider** component provides terrain data to the physics system in the form of a heightfield and material assignments.  
+You can configure the dimensions of the collider by modifying the [Axis Aligned Box Shape Component](/docs/user-guide/components/reference/shape/axis-aligned-box-shape) on the same entity.
+The **Box component** is required for the **Terrain Physics Collider** to operate.
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md:10]:
```suggestion
```

Suggest moving this to a Dependencies heading to match style of other components.

File: content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+## Usage ##
+
+You can assign which materials are assigned to surfaces, by selecting a surface type in the surface pull down menu, then selecting a physics material type in the material pull down.
+
+## Provider ##
+
+[Terrain Gem](/docs/user-guide/gems/reference/environment/terrain)
+
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md:19]:
```suggestion

## Dependencies

The **Box component** is required for the **Terrain Physics Collider** to operate.

```

File: content/docs/user-guide/gems/reference/environment/terrain/_index.md [hunk 7]
```diff
@@ -29,5 +29,6 @@ To enable the Terrain Gem, do the following:
 
 | Component | Description | 
 | - | - |
+| [Terrain Physics Collider](/docs/user-guide/components/reference/terrain/terrain_physics_collider/) | Provides terrain data to a physics collider in the form of a heightfield and surface to material mapping. |
```
Comment (FiniteStateGit) [inline content/docs/user-guide/gems/reference/environment/terrain/_index.md:32]:
```suggestion
| [Terrain Physics Collider](/docs/user-guide/components/reference/terrain/terrain_physics_collider) | Provides terrain data to a physics collider in the form of a heightfield and surface to material mapping. |
```
Links to directories should have the trailing slash, but it's not needed for pages in the `terrain` directory.  Also this link may need to be changed to use hyphens.

File: content/docs/user-guide/gems/reference/environment/terrain/_index.md [hunk 8]
```diff
@@ -29,5 +29,6 @@ To enable the Terrain Gem, do the following:
...
 | [Terrain Layer Spawner](/docs/user-guide/components/reference/terrain/layer_spawner.md) | Spawns a terrain region contained within configurable bounds, and allows prioritization of overlapping terrain layers. |
```
Comment (FiniteStateGit) [inline content/docs/user-guide/gems/reference/environment/terrain/_index.md:33]:
```suggestion
| [Terrain Layer Spawner](/docs/user-guide/components/reference/terrain/layer_spawner) | Spawns a terrain region contained within configurable bounds, and allows prioritization of overlapping terrain layers. |
```

File: content/docs/user-guide/gems/reference/environment/terrain/_index.md [hunk 9]
```diff
@@ -29,5 +29,6 @@ To enable the Terrain Gem, do the following:
...
-| [Terrain Height Gradient List](/docs/user-guide/components/reference/terrain/terrain_height_gradient_list.md) | Provides terrain height data from a list of gradients. |
\ No newline at end of file
+| [Terrain Height Gradient List](/docs/user-guide/components/reference/terrain/terrain_height_gradient_list.md) | Provides terrain height data from a list of gradients. |
```
Comment (FiniteStateGit) [inline content/docs/user-guide/gems/reference/environment/terrain/_index.md:34]:
```suggestion
| [Terrain Height Gradient List](/docs/user-guide/components/reference/terrain/terrain_height_gradient_list) | Provides terrain height data from a list of gradients. |
```

**Review by FiniteStateGit — CHANGES_REQUESTED** [review]
I wasn't sure about the filename for terrain_physics_collider.md, all the other pages I've seen use hyphens.  It's a recommendation from Google for indexing purposes.

File: content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md (hunk 5 above)
Comment (sphrose) [reply content/docs/user-guide/components/reference/terrain/terrain_physics_collider.md:10]: Done

File: content/docs/user-guide/components/reference/terrain/terrain-physics-collider.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+Type: `AZStd::Vector2`  
+The resolution of the heightfield grid.
+
+### GetHeightfieldGridSize
+
+Retrieves the size of the heightfield in the form of a row and column count.
+
+#### Parameters
+
+#### `NumColumns \[out\]`
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/terrain/terrain-physics-collider.md:46]:
```suggestion
#### `NumColumns [out]`
```

File: content/docs/user-guide/components/reference/terrain/terrain-physics-collider.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+
+Type: `int32_t`  
+The number of columns in the heightfield.
+
+#### `NumRows \[out\]`
```
Comment (FiniteStateGit) [inline content/docs/user-guide/components/reference/terrain/terrain-physics-collider.md:51]:
```suggestion
#### `NumRows [out]`
```

**Review by FiniteStateGit — CHANGES_REQUESTED** [review]
Just a minor edit.

**Comment by sptramer** [comment]
Closing as stale; this was an instructional PR and needs to be reworked.

**Comment by sptramer** [comment]
I was very tired and closed the wrong pull request!!!!

File: content/docs/user-guide/components/reference/terrain/terrain-physics-collider.md (hunk 10 above)
Comment (sphrose) [reply content/docs/user-guide/components/reference/terrain/terrain-physics-collider.md:46]: Fixed

File: content/docs/user-guide/components/reference/terrain/terrain-physics-collider.md (hunk 11 above)
Comment (sphrose) [reply content/docs/user-guide/components/reference/terrain/terrain-physics-collider.md:51]: Fixed

**Review by FiniteStateGit — APPROVED** [review]
Looks good.  Approved.

**Review by willihay — APPROVED** [review]

**Pull request merged**